### DISCUSSION TITLE
Can I generate some residual EV charing data?

### DISCUSSION THREAD

**Discussion by b-y-f** [zach401/acnportal#113] [Q&A] [answered]
Can I generate some residual EV charging data?
I know those data from public stations, but we desperately need some fake household EV charging data.

**Comment by sunash** [comment, marked as answer]
Unfortunately, we don't have mechanisms to generate fake residential EV charging data. Of course, you are free to generate your own residential EV charging data from a residential data source of your choosing and use this data with acnportal. If you use a Gaussian mixture model to generate residential EV data, section 4 of this paper may be helpful: https://ev.caltech.edu/assets/pub/ACN_Data_Analysis_and_Applications.pdf

Events generated from a distribution are of this class in the Simulator:
https://github.com/zach401/acnportal/blob/master/acnportal/acnsim/events/stochastic_events.py

And make use of this ChargingNetwork subclass:
https://github.com/zach401/acnportal/blob/master/acnportal/contrib/acnsim/network/stochastic_network.py

However, if you a dealing with a residential simulation, you might have the same number of cars as chargers, in which case you can just assign a station to each EV and use the events.py classes and the base ChargingNetwork.

**Reply by b-y-f** [reply]
Thanks, We only got <5 half-hourly household data samples, not like acnportal with 3 car parks, can I still do the simulation?

**Reply by sunash** [reply]
You can do the simulation, but you would need to make your own ChargingNetwork instances with fewer charging stations, and an EventQueue that describes residential charging. Also, keep in mind that Events in ACN-Sim require Plugin and Unplug times, so you might need to infer those from your data.

**Reply by b-y-f** [reply]
Hi @sunash , all we got is 3 datasets of charging patterns from different house with different EV, the format is like 

```
id | ts| ev_charging_value
-- | -- | -- | --
...
23 | 2019-06-01 11:30:00 | 0.000
24 | 2019-06-01 12:00:00 | 0.000
25 | 2019-06-01 12:30:00 | 0.000
26 | 2019-06-01 13:00:00 | 0.000
27 | 2019-06-01 13:30:00 | 0.000
28 | 2019-06-01 14:00:00 | 0.000
29 | 2019-06-01 14:30:00 | 1327.999
30 | 2019-06-01 15:00:00 | 1857.003
31 | 2019-06-01 15:30:00 | 1854.000
32 | 2019-06-01 16:00:00 | 1854.000
33 | 2019-06-01 16:30:00 | 1854.000
34 | 2019-06-01 17:00:00 | 701.000
35 | 2019-06-01 17:30:00 | 0.000
36 | 2019-06-01 18:00:00 | 0.000
37 | 2019-06-01 18:30:00 | 0.000
...
500K rows
```

Can we feed `acn-sim` with only three of those datasets?

**Reply by sunash** [reply]
You can: make a PluginEvent when the charging rate goes from zero to nonzero, and then an UnplugEvent when the charging rate drops to zero again. The energy demanded can just be the sum of charging rates over the charging session. Then you can run ACN SIm with the arrivals, departures, and energy demands for the charging network associated with this data.

That being said, it might be helpful to know what you are trying to study with this data. ACN-Sim might not be the best tool for your purposes, as it would not make use of the charging time series you have beyond calculating total energy demand for each session. If you're doing studies on required infrastructure or different smart charging algorithms' performances, ACN-Sim would be helpful.

**Comment by tpolong** [comment]
您好，您的邮件已收到，谢谢！
